nifi-extension-bundles/nifi-extension-utils/nifi-file-transfer/src/main/java/org/apache/nifi/processor/util/file/transfer/FileTransferConflictUtil.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.processor.util.file.transfer;
+
+import org.apache.nifi.flowfile.FlowFile;
+import org.apache.nifi.logging.ComponentLog;
+
+import java.io.IOException;
+import java.util.UUID;
+
+public final class FileTransferConflictUtil {
+    private FileTransferConflictUtil() {
+    }
+
+    /**
+     * Generates a unique filename by using a UUID prefix.
+     * This approach virtually eliminates the possibility of name collisions without requiring multiple remote file checks.
+     */
+    public static String generateUniqueFilename(final FileTransfer transfer,
```

Comment:
As this is only used once in `FetchFileTransfer`, recommend moving it to a private method in that class and removing this utility class.